storagenode/gracefulexit/chore_test.go

```diff
@@ -4,52 +4,204 @@
+
+	// run the satellite chore to build the transfer queue.
+	satellite1.GracefulExit.Chore.Loop.TriggerWait()
+
+	// check that the satellite knows the storage node is exiting.
+	exitingNodes, err := satellite1.DB.OverlayCache().GetExitingNodes(ctx)
+	require.NoError(t, err)
+	require.Len(t, exitingNodes, 1)
+	require.Equal(t, exitingNode.ID(), exitingNodes[0])
+
+	queueItems, err := satellite1.DB.GracefulExit().GetIncomplete(ctx, exitStatus.NodeID, 10, 0)
+	require.NoError(t, err)
+	require.Len(t, queueItems, 1)
+
+	// run the SN chore again to start processing transfers.
+	exitingNode.GracefulExit.Chore.Loop.TriggerWait()
+
+	// check that there are no more items to process
+	queueItems, err = satellite1.DB.GracefulExit().GetIncomplete(ctx, exitStatus.NodeID, 10, 0)
+	require.NoError(t, err)
+	require.Len(t, queueItems, 0)
+
+	exitProgress, err = exitingNode.DB.Satellites().ListGracefulExits(ctx)
+	require.NoError(t, err)
+	for _, progress := range exitProgress {
+		if progress.SatelliteID == satellite1.ID() {
+			require.NotNil(t, progress.FinishedAt)
+		}
+	}
+
+	// make sure there are no more pieces on the node.
+	namespaces, err := exitingNode.DB.Pieces().ListNamespaces(ctx)
+	require.NoError(t, err)
+	for _, ns := range namespaces {
+		err = exitingNode.DB.Pieces().WalkNamespace(ctx, ns, func(blobInfo storage.BlobInfo) error {
+			return errs.New("found a piece on the node. this shouldn't happen.")
+		})
+		require.NoError(t, err)
+	}
+}
+func getNodePieceCounts(ctx context.Context, planet *testplanet.Planet) (_ map[storj.NodeID]int, err error) {
+	nodePieceCounts := make(map[storj.NodeID]int)
+	for _, n := range planet.StorageNodes {
+		node := n
+		// make sure there are no more pieces on the node.
+		namespaces, err := node.DB.Pieces().ListNamespaces(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, ns := range namespaces {
+			err = node.DB.Pieces().WalkNamespace(ctx, ns, func(blobInfo storage.BlobInfo) error {
+				nodePieceCounts[node.ID()]++
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return nodePieceCounts, err
+}
+
+// findNodeToExit selects the node storing the most pieces as the node to graceful exit.
+func findNodeToExit(ctx context.Context, planet *testplanet.Planet, objects int) (*storagenode.Peer, error) {
+	satellite := planet.Satellites[0]
+	keys, err := satellite.Metainfo.Database.List(ctx, nil, objects)
+	if err != nil {
+		return nil, err
+	}
+
+	pieceCountMap := make(map[storj.NodeID]int, len(planet.StorageNodes))
+	for _, sn := range planet.StorageNodes {
+		pieceCountMap[sn.ID()] = 0
+	}
+
+	for _, key := range keys {
+		pointer, err := satellite.Metainfo.Service.Get(ctx, string(key))
+		if err != nil {
+			return nil, err
+		}
+		pieces := pointer.GetRemote().GetRemotePieces()
+		for _, piece := range pieces {
+			pieceCountMap[piece.NodeId]++
+		}
+	}
+
+	var exitingNodeID storj.NodeID
+	maxCount := 0
+	for k, v := range pieceCountMap {
```

Comment:
nit - it seems like we could also use `getNodePieceCounts` here, but at the same time I think it's cool to have two methods of calculating piece counts per node in this file (going through individual nodes vs. going through metainfo)